src/main/java/com/linkedin/kmf/services/ConsumeService.java

```diff
@@ -181,13 +205,24 @@ private void consume() throws Exception {
         nextIndexes.put(partition, index + 1);
         _sensors._recordsLost.record(index - nextIndex);
       }
+
+      Broker broker = _partitionToBroker.get(partition);
+      Collection<EndPoint> endPoints = scala.collection.JavaConversions.asJavaCollection(broker.endPoints());
+      for (EndPoint endpoint : endPoints) {
+        String brokerUrl = endpoint.host() + ":" + endpoint.port();
+        if (!_sensors._delayPerBroker.containsKey(brokerUrl)) {
+          _sensors.addBrokerSensors(brokerUrl);
+        }
+        _sensors._delayPerBroker.get(brokerUrl).record(currMs - prevMs);
+      }
     }
   }
 
   @Override
   public synchronized void start() {
     if (_running.compareAndSet(false, true)) {
       _thread.start();
+      _handlePartitionLeaderInfoExecutor.scheduleWithFixedDelay(new PartitionLeaderInfoHandler(), 1000, 10000, TimeUnit.MILLISECONDS);
```

Comment:
@mpetazzoni Delay is 1000 because, if a topic is not present then TopicManagementService creates a topic and then consumer start: https://github.com/linkedin/kafka-monitor/blob/master/src/main/java/com/linkedin/kmf/services/ConsumeService.java#L134

That's why I put 1000 as initial dealy. If the topic is not present then there will be no partition leader assignment 

I agree with the interval, I have made it configurable with the new commit.